src/univariate/affine.jl

```diff
@@ -0,0 +1,182 @@
+@deprecate LocationScale(args...; kwargs...) affine(args...)
+
+"""
+    affine(shift, scale, base_dist)
+
+Return a shifted and scaled (affinely transformed) version of `base_dist`.
+
+If ``Z`` is a random variable with distribution `base_dist`, then `affine` returns the 
+distribution of the random variable
+```math
+X = shift + scale * Z
+```
+
+If `base_dist` is a discrete distribution, the probability mass function of the transformed 
+distribution is given by
+```math
+P(X = x) = P\\left(Z = \\frac{x-shift}{scale} \\right).
+```
+If `base_dist` is a continuous distribution, the probability density function of
+the transformed distribution is given by
+```math
+f(x) = \\frac{1}{scale} base_dist \\! \\left( \\frac{x-shift}{scale} \\right).
+```
+
+If `base_dist` falls under a location-scale family, `affine` will attempt to return a 
+transformed distribution of type `base_dist` if an `affine` method has been implemented. 
+Otherwise, it will fall back on the `Affine` wrapper type.
+
+```julia
+affine(shift, scale, base_dist)  # location-scale transformed distribution
+params(d)            # Get the parameters, i.e. (shift, scale, base_dist)
+location(d)          # Get the location parameter
+scale(d)             # Get the scale parameter
+```
+
+External links
+[Location-Scale family on Wikipedia](https://en.wikipedia.org/wiki/Location%E2%80%93scale_family)
+"""
+struct Affine{T<:Real, S<:ValueSupport, D<:UnivariateDistribution{S}} <: UnivariateDistribution{S}
+    shift::T
+    scale::T
+    base_dist::D
+    function Affine{T,S,D}(shift::T, scale::T, base_dist::D; check_args=true) where {
+        T<:Real, S<:ValueSupport, D<:UnivariateDistribution{S}
+    }
+        !check_args || @check_args(Affine, !iszero(scale))
+        return new{T,S,D}(shift, scale, base_dist)
+    end
+end
+
+function affine(shift::Real, scale::Real, base_dist::UnivariateDistribution)
+    T = promote_type(eltype(base_dist), typeof(shift), typeof(scale))
+    D = typeof(base_dist)
+    S = value_support(D)
+    return Affine{T,S,D}(T(shift), T(scale), base_dist)
+end
+
+# Composing affine transformations
+function affine(shift::Real, scale::Real, d::Affine)
+    return affine(shift + scale * d.shift, scale * d.scale, d.base_dist)
+end
```

Comment:
I suggest removing this:

```suggestion
```

We already have an API and should just use `*`, `+`, etc.

Reply:
`*`, `+`, etc. are implemented as `affine` using args 0/1; is there any reason not to keep it at least internally?

Reply:
It might actually be worth exporting to let people do an affine transformation all at once, instead of having to separately create 2 structs.

And keeping it makes it easier to overload the function, since someone can just overload `affine` instead of overloading each of `+`.

Reply:
I disagree, I don't think it is useful. Nor should it be exported. A main reason is that `affine` is really a too common name and should not be owned by Distributions. Another reason is that it is much easier to implement specializations for certain distributions if one can implement the shifting and scaling separately.